File: 기장지도/check_gwangju.mjs
import fs from 'fs';

const envContent = fs.readFileSync('.env.local', 'utf-8');
let SUPABASE_URL = '';
let SUPABASE_KEY = '';

envContent.split('\n').forEach(line => {
  if (line.startsWith('VITE_SUPABASE_URL=')) SUPABASE_URL = line.split('=')[1].trim();
  if (line.startsWith('VITE_SUPABASE_ANON_KEY=')) SUPABASE_KEY = line.split('=')[1].trim();
});

async function checkGwangju() {
  const url = `${SUPABASE_URL}/rest/v1/churches?address=like.*광주*&select=name,address,lat,lng`;
  
  const response = await fetch(url, {
    headers: {
      'apikey': SUPABASE_KEY,
      'Authorization': `Bearer ${SUPABASE_KEY}`
    }
  });
  
  const data = await response.json();
  console.log(`Total DB entries containing 광주: ${data.length}`);
  
  const no_coords = data.filter(c => !c.lat || !c.lng);
  console.log(`Without coordinates: ${no_coords.length}`);
  if (no_coords.length > 0) console.log("Sample without coords:", no_coords.slice(0, 3));
  
  const with_coords = data.filter(c => c.lat && c.lng);
  if (with_coords.length > 0) console.log("Sample WITH coords:", with_coords.slice(0, 3));
}

checkGwangju();

File: server/check_gwangju.py
import os
from supabase import create_client

url = os.environ.get('SUPABASE_URL', '')
key = os.environ.get('SUPABASE_KEY', '')

# If no env vars, try reading from .env
if not url:
    try:
        with open('.env', 'r', encoding='utf-8') as f:
            for line in f:
                if line.startswith('SUPABASE_URL='): url = line.strip().split('=')[1]
                if line.startswith('SUPABASE_KEY='): key = line.strip().split('=')[1]
    except Exception as e: print(e)

if url and key:
    supabase = create_client(url, key)
    res = supabase.table('churches').select('name, address, lat, lng').like('address', '%광주%').execute()
    
    gwangju_churches = res.data
    print(f"Total containing 광주: {len(gwangju_churches)}")
    
    no_coords = [c for c in gwangju_churches if not c.get('lat') or not c.get('lng')]
    print(f"Without coordinates: {len(no_coords)}")
    print("Sample with no coords:")
    for c in no_coords[:5]:
        print(c)
    
    with_coords = [c for c in gwangju_churches if c.get('lat') and c.get('lng')]
    print("Sample with coords:")
    for c in with_coords[:5]:
        print(c)
